api/src/main/java/org/openmrs/module/serialization/xstream/XStreamSerializer.java

```diff
@@ -161,17 +161,16 @@ protected MapperWrapper wrapMapper(MapperWrapper next) {
 		xstream.setMarshallingStrategy(new CustomReferenceByIdMarshallingStrategy());
 	}
 
-    @PostConstruct
-    private void init(){
+	@Override
+	public void afterPropertiesSet() throws Exception {
```

Comment:
Can you explain a bit as to why we need to make this change?

Reply:
So when we changed the scope of javax.annotation-api bunch of tests started to fail on Java 17. After investigating, I noticed this caused a difference in Java 8 and 17 serialization output.

We only use javax.annotation-api here: https://github.com/openmrs/openmrs-module-serialization.xstream/blob/d01a5c0eb76d3a0fb77dd7265332db47083b5a73/api/src/main/java/org/openmrs/module/serialization/xstream/XStreamSerializer.java#L164

So I switched to the alternative of @PostConstruct and used afterPropertiesSet() method.

Reply:
Hope this clears things up